adi/cn0554.py

```diff
@@ -0,0 +1,148 @@
+            in_name = "voltage"
+            for char in in_ch.name:
+                if char.isnumeric():
+                    in_name += char
+
+            setattr(
+                self, in_name + "_in_range", self._in_range_available[0]
+            )  # default range of HW
+
+            self.adc_in_channels.append(in_name)
+
+        self.dac = adi.ltc2688(uri=uri)
+        self.dac_out_channels = self.dac.channel_names
+        self.out_reference = self._out_reference_available[0]
+
+    @property
+    def out_type(self):
+        """ Get value of CN0554's rx output type (raw or SI) """
+        return self.adc._rx_output_type
+
+    @out_type.setter
+    def out_type(self, val):
+        """ Set value of CN0554's rx output type (raw or SI) """
+        self.adc._rx_output_type = val
+
+    @property
+    def enabled_in_channels(self):
+        """ Get list of enabled input adc channels """
+        return self.adc.rx_enabled_channels
+
+    @enabled_in_channels.setter
+    def enabled_in_channels(self, val):
+        """ Set list of enabled input adc channels """
+        self.adc.rx_enabled_channels = val
+
+    @property
+    def out_channels(self):
+        """ Get number of DAC output channels """
+        return self._out_channels
+
+    @property
+    def in_channels(self):
+        """ Get number of ADC input channels """
+        return self._in_channels
+
+    @property
+    def out_reference(self):
+        """ Get voltage reference used for DAC channels """
+        return self.out_reference
+
+    @out_reference.setter
+    def out_reference(self, value):
+        """ Set voltage reference used for DAC channels """
+        if value in self._out_reference_available:
+            self.dac.vref = value
+
+    @property
+    def sample_rate(self):
+        """ Get CN0554's AD7124 sampling rate """
+        return self.adc.sample_rate
+
+    @sample_rate.setter
+    def sample_rate(self, value):
+        """ Set CN0554's AD7124 sampling rate """
+        self.adc.sample_rate = value
+
+    @property
+    def data_points(self):
+        """ Get number of datapoints used during capture """
+        return self.adc.rx_buffer_size
+
+    @data_points.setter
+    def data_points(self, value):
+        """ Set number of datapoints used during capture """
+        self.adc.rx_buffer_size = value
+
+    def get_data(self):
+        """ Get data from enabled ADC channels """
+        return self.adc.rx()
+
+    def convert_to_volts(self, in_voltage, channel):
+        """ Convert ADC data using scale factors based on header configuration """
+        ch_range = eval("self." + str(channel) + "_in_range")
+        if ch_range != self._in_range_available[2]:
+            scale_sig = []
+            for sigs in in_voltage:
+                scale_sig.append(sigs * self.scale_res)
+            return scale_sig
+
+        elif ch_range == self._in_range_available[2]:
+            return in_voltage
+
+        else:
+            return None
```

Comment:
This case should generate an error. And the "else" is not required since the other cases return

Reply:
rewrote function to raise an error if an invalid in_range attribute value is detected based on index look up.